### PR TITLE
feat: create mappers functions for file paths

### PR DESCRIPTION
**Pull request by AlexTheWizardL**
This pull request covers 2 functions:

1. `file_id_to_path()`
2. `file_path_to_id()`

This functions will help to easily find id or path in the cached files.

@lasuk Please review the changes and provide feedback or approval for merging.

### CODE REVIEW

**Comment by lasuk** [comment]
I've added two commits:
- bbca34bc with minor modifications to prettify code and fix checks
- e5038436 to improve the logic of pytest fixtures.

@AlexTheWizardL: I've left a TODO in the code. Can you address this (restore files on the CashCtrl server to the initial state before the test startet)? Thanks
```
# TODO: We should restore original files after the test is complete
```

**Comment by lasuk** [comment]
Let's also override below methods in `CashedCashCtrlClient`. The overridden methods will firs call the parent class method and then invalidate the cache:
```python
mirror_directory()
upload_file()
update_categories()
```

File: cashctrl_api/cashed_client.py [hunk 1]
```diff
@@ -338,6 +379,90 @@ def tax_code_to_id(self, name: str, allow_missing = False) -> int | None:
...
+        categories (folders) on the server to match the local structure.
+
+        Parameters:
+            directory (str | Path): Path of the local directory to mirror.
+            delete_files (bool, optional): If True, deletes remote files
+                without a corresponding local file. Also empties the recycle
+                bin to release references and allow for category deletion.
+            delete_categories (bool, optional): If True, deletes unused
+                categories (folders) on the server.
+        """
```
[the remark this answers is not in the document]
Comment (lasuk) [reply cashctrl_api/cashed_client.py:401]: Also applies to `upload_file` and `update_categories`.

File: cashctrl_api/client.py [hunk 2]
```diff
@@ -262,7 +262,7 @@ def update_categories(self, resource: str, target: Dict[str, int] | List[str],
             if resource == 'account':
                 target_series = pd.Series(target.keys())
             else:
-                target_series = pd.Series(target).unique()
+                target_series = pd.Series(pd.Series(target).unique())
```
Comment (lasuk) [inline cashctrl_api/client.py:265]: Why do we need this change?

File: tests/test_cached_files.py [hunk 3]
```diff
@@ -8,21 +8,96 @@
...
+    (nested_subdir / 'file4.txt').write_text("File in the nested directory.")
+    (nested_subdir / 'file5.log').write_text("Another nested directory file.")
+    return tmp_path
+
+@pytest.fixture(scope="module")
+def cc_client(mock_directory):
+    """Create a CachedCashCtrlClient, populate with files and folders."""
     cc_client = CachedCashCtrlClient()
+    initial_files = cc_client.list_files()
+    # check is this needed
```
Comment (lasuk) [inline tests/test_cached_files.py:34]: I believe the comment is outdated.

File: tests/test_cached_files.py [hunk 4]
```diff
@@ -8,21 +8,96 @@
...
+    cc_client.mirror_directory(mock_directory, delete_files=False)
+
+    # We create a fresh instance with empty cache, because the cache is
+    # populated when mirroring a directory
+    cc_client = CachedCashCtrlClient()
+
+    yield cc_client
+
+    files = cc_client.list_files()
+    # check is this needed
```
Comment (lasuk) [inline tests/test_cached_files.py:44]: I don't understand the comment.

File: tests/test_cached_files.py [hunk 5]
```diff
@@ -8,21 +8,96 @@
...
+    to_delete = set(files['id']).difference(initial_files['id'])
+    params = { 'ids': ','.join(str(i) for i in to_delete), 'force': True }
+    cc_client.post("file/delete.json", params=params)
+    # Empty recycle bin to release references before category deletion
```
Comment (lasuk) [inline tests/test_cached_files.py:48]: Let's be less invasive and leave the recycle bin untouched. It seems that just deleting it is not sufficient to delete file categories.

File: tests/test_cached_files.py [hunk 6]
```diff
@@ -8,21 +8,96 @@
...
+    # check is this needed
+    cc_client.mirror_directory(mock_directory, delete_files=False)
+
+    # We create a fresh instance with empty cache, because the cache is
+    # populated when mirroring a directory
+    cc_client = CachedCashCtrlClient()
+
+    yield cc_client
+
+    files = cc_client.list_files()
```
Comment (lasuk) [inline tests/test_cached_files.py:43]: Please add a comment explaining this code block (e.g. `# Delete files added in the test`)

**Review by lasuk — COMMENTED** [review]
Thanks for the modifications. I left a few comments. Back to you @AlexTheWizardL.

File: cashctrl_api/cashed_client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexTheWizardL) [reply cashctrl_api/cashed_client.py:401]: I see docstrings are auto-inherited. So no need to do additional work
Screenshot 2024-06-17 at 13 16 32

File: cashctrl_api/cashed_client.py [hunk 7]
```diff
@@ -338,6 +380,36 @@ def tax_code_to_id(self, name: str, allow_missing = False) -> int | None:
...
+                    id: int | str | None = None,
+                    name: str | None = None,
+                    category: int | str | None = None,
+                    mime_type: str | None = None) -> int:
+        super().upload_file(file=file,
+                            id=id,
+                            name=name,
+                            category=category,
+                            mime_type=mime_type)
+        self.invalidate_files_cache()
```
Comment (lasuk) [inline cashctrl_api/cashed_client.py:402]: What do you think of this shorter alternative?
```python
def upload_file(self, *args, **kwargs) -> int:
    super().upload_file(*args, **kwargs)
    self.invalidate_files_cache()
```

**Review by lasuk — COMMENTED** [review]
Thanks for the modifications. Two comments remain. Back to you @AlexTheWizardL .

File: cashctrl_api/cashed_client.py [hunk 8]
```diff
@@ -338,6 +380,22 @@ def tax_code_to_id(self, name: str, allow_missing = False) -> int | None:
...
+
+    def update_categories(self, *args, **kwargs):
+        super().update_categories(*args, **kwargs)
+        resource = kwargs.get('resource', None)
+
+        if resource == 'file':
+            self.invalidate_files_cache()
+        elif resource == 'account':
+            self.invalidate_account_categories_cache()
```
Comment (lasuk) [inline cashctrl_api/cashed_client.py:398]: I guess the standard way to access the first argument is:
```python
    def update_categories(self, resource: str, *args, **kwargs):
        super().update_categories(resource, *args, **kwargs)
        if resource == 'file':
            self.invalidate_files_cache()
        elif resource == 'account':
            self.invalidate_account_categories_cache()
```

**Review by lasuk — COMMENTED** [review]
Thanks. One more comment. Back to you @AlexTheWizardL.

**Comment by lasuk** [comment]
Thanks! Merging.

**Pull request merged**